src/hooks/useWindowSpecs.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react'
+
+const useWindowSpecs = (): { innerWidth: number; innerHeight: number } => {
+  const isClient = typeof window === 'object'
+
+  function getSize(): { innerWidth: number | undefined; innerHeight: number | undefined } {
+    return {
+      innerWidth: isClient ? window.innerWidth : undefined,
+      innerHeight: isClient ? window.innerHeight : undefined,
+    }
+  }
+
+  const [innerWindowSpecs, setInnerWindowSpecs] = useState(getSize)
+  function handleResize(): void | number {
+    // check that client can handle this sexiness
+    const innerWidth = typeof window !== 'object' ? null : window.innerWidth
+    const innerHeight = typeof window !== 'object' ? null : window.innerHeight
```

Comment:
Why not use `getSize()` here? AFAICT, lines 16,17 are just a duplicate of 8,7.

Reply:
good call, forgot about optimising it, thanks